### PR TITLE
Fix rotation stuck in portrait on iPad mini

### PR DESCRIPTION
**Pull request by samscherer**
-Remove forced rotation since it is breaking on older devices
-Allow iOS 9.3

This stops the app from being stuck in portrait on first generation iPad Mini and probably some other iPads. If someone can check the iPad 2 and iPad 3 which max out on iOS 9 let me know if it works.

I removed the orientation lock on the stats view. I couldn't find any reason why it should be locked into portrait. It does look a bit cramped on an iPhone 5s in testing, but perhaps better to have the option than to not? Dexcom G6 for ios does this also and i find it a bit intrusive.

### CODE REVIEW

**Comment by dhermanns** [comment]
Hi Sam,

I think the Stats view looks really cramped in portrait mode, even on bigger iPhones.

We have not many iPad users. I think really few on older iOS Versions. So I‘m not sure if we should cramp the screens for all others to get it seamless running on old iPads.

@florianpreknya : what do you think?

**Comment by samscherer** [comment]
Maybe I can find a fix that still forces the rotation for stats. I have seen other apps do this, like the G6 app.

The way currently I use this app is to run it overnight on an iPad. I use the older devices since doing that will eventually burn in the screen. I thought maybe others were doing that too.

**Comment by florianpreknya** [comment]
I would permit free orientation only for ipads.
My proposal is to check the device user interface idiom, executing the orientation code only for phones. An example:

```Swift
override var supportedInterfaceOrientations : UIInterfaceOrientationMask {
        if UIDevice.current.userInterfaceIdiom == .phone {
            return UIInterfaceOrientationMask.portrait
        } else {
            return super.supportedInterfaceOrientations
        }
    }
````

Is this a solution for you, @samscherer ?

**Comment by imakeinterfaces** [comment]
It should work, but we still have to force the viewport back to portrait on `viewDidAppear`. Maybe we can instead keep track of the previous view type and only rotate if it was stats. I don't like the idea of forcing a noop. Also it seems like the old code forced portrait, and then back to landscape. I'll comment on that in a line comment.

File: nightguard/StatsViewController.swift [hunk 1]
```diff
@@ -31,14 +31,10 @@ class StatsViewController: UIViewController {
     }
     
     override func viewDidAppear(_ animated: Bool) {
+
         super.viewDidAppear(animated)
         
-        // force the display into horizontal orientation
-        UIDevice.current.setValue(UIInterfaceOrientation.portrait.rawValue, forKey: "orientation")
-        UIDevice.current.setValue(UIInterfaceOrientation.landscapeRight.rawValue, forKey: "orientation")
```
Comment (imakeinterfaces) [inline nightguard/StatsViewController.swift]: Not sure why it sets portrait and then landscape. Maybe this is fixing an edge case bug that I'm unable to reproduce?

Comment (florianpreknya) [reply nightguard/StatsViewController.swift]: Maybe @dhermanns can give a little help with these lines...

**Comment by florianpreknya** [comment]
The idea was to keep the ipad version without forcing any orientation, so it will be portrait when holding the ipad in portrait mode, landscape otherwise, changing freely orientations when the user rotates the device. That is achieved now with your code. 
For phones, we should keep the original code logic.

Also, if something doesn't work on older iOS versions, we ca write different code there:
```swift
if #available(iOS 10, *) {
    // current code
} else {
    // iOS 9 code
}
```

Am I still missing something? Unfortunately I cannot test it on a real iOS 9 device, but will do in simulator if it can help.

**Comment by dhermanns** [comment]
I'm not sure whether this is still an issue, due to the changes florian implemented. But since this PR is pretty outdated - feel free to add comments. But closing it for now.

**Pull request closed without merging**